### PR TITLE
17220 code organization

### PR DESCRIPTION
**Pull request by rianfowler**
Addresses: 

- [Document how VA.gov maps to our application source code](https://github.com/department-of-veterans-affairs/vets.gov-team/issues/15777)
- [Create 'Code Organization' document for FE code](https://github.com/department-of-veterans-affairs/vets.gov-team/issues/17220)

### CODE REVIEW

File: packages/documentation/src/pages/getting-started/bundles-and-code-organization.mdx [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+    - `manifest-loader.js` - passes front matter of content files to app entry files
+	  - `parse-app-settings.js` - reads settings.js - a file containing app information generated during Metalsmith build
+- `/docs` - best practices and troubleshooting docs
+- `/jenkins` - jenkins setup, build, deploy script for `vets-websites` repo
+- `/script`
+  - `/custom-eslint` - project specific linting plugin
+  - `/hooks` - -git -precommit hook
+  - `app-list.sh` - bash script that outputs a list of the applications in vets-website
+  - `build.js` - imports and runs build script in src/site/stages/build
+  - `csv-path-builder.js` - TODO
```
Comment (rianfowler) [inline packages/documentation/src/pages/getting-started/bundles-and-code-organization.mdx]: not sure what this does

File: packages/documentation/src/pages/getting-started/bundles-and-code-organization.mdx [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+  - `heroku-postbuild.sh` - TODO - content deploy to heroku
```
Comment (rianfowler) [inline packages/documentation/src/pages/getting-started/bundles-and-code-organization.mdx]: not sure what this does

File: packages/documentation/src/pages/getting-started/bundles-and-code-organization.mdx [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+  - `mocha.js` - unit test runner and watch task
+  - `prearchive.js` - imports and run prearchive in src/site/stages/prearchive TODO
```
Comment (rianfowler) [inline packages/documentation/src/pages/getting-started/bundles-and-code-organization.mdx]: not sure what this is used for

**Comment by rianfowler** [comment]
guess I forgot to add reviewers to this

File: packages/documentation/src/pages/getting-started/bundles-and-code-organization.mdx (hunk 3 above)
Comment (jbalboni) [reply packages/documentation/src/pages/getting-started/bundles-and-code-organization.mdx]: This updates the paths to js, css, and image assets to point to the S3 asset bucket, our makeshift CDN.

File: packages/documentation/src/pages/getting-started/bundles-and-code-organization.mdx [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+  - `run-mockapi.sh` - runs js mock vets-app
+  - `run-nightwatch.sh` - test harness for running end to end tests
+  - `run-puppeteer-tests.sh` - test harness for puppeteer tests (proof of concept)
+  - `run-unit-test.sh` - runs unit tests
+  - `security-check.js` - checks for security advisories in dependencies
+  - `update-json-schema.js` - helper for updating dependency version of vets-jason-schema
+  - `watch.js` - watch task for unit tests (proof of concept)
+- `/src` - source files for vets-website web applications
+  - `/applications` - application specific code that's into -entry.js files
+  - `/platform` - shared code that's bundled into vendor.js, injected directly into the site, or used for testing
```
Comment (jbalboni) [inline packages/documentation/src/pages/getting-started/bundles-and-code-organization.mdx]: Maybe worth clarifying, there's not any real code from /platform that's in the vendors, just a file that imports polyfills.

File: packages/documentation/src/pages/getting-started/bundles-and-code-organization.mdx [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+  - `/custom-eslint` - project specific linting plugin
+  - `/hooks` - -git -precommit hook
+  - `app-list.sh` - bash script that outputs a list of the applications in vets-website
+  - `build.js` - imports and runs build script in src/site/stages/build
+  - `csv-path-builder.js` - TODO
+  - `heroku-postbuild.sh` - TODO - content deploy to heroku
+  - `mocha.js` - unit test runner and watch task
+  - `prearchive.js` - imports and run prearchive in src/site/stages/prearchive TODO
+  - `prebuild.js` - npm prebuild script that checks node version and install precommit hooks
+  - `preview.js` - metalsmith build script for drupal preview
```
Comment (jbalboni) [inline packages/documentation/src/pages/getting-started/bundles-and-code-organization.mdx]: Might be better described as "Express server used to preview content pages on demand from the Drupal admin site"

File: packages/documentation/src/pages/getting-started/bundles-and-code-organization.mdx (hunk 1 above)
Comment (jbalboni) [reply packages/documentation/src/pages/getting-started/bundles-and-code-organization.mdx]: No idea either

File: packages/documentation/src/pages/getting-started/bundles-and-code-organization.mdx [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+  - `run-nightwatch.sh` - test harness for running end to end tests
+  - `run-puppeteer-tests.sh` - test harness for puppeteer tests (proof of concept)
+  - `run-unit-test.sh` - runs unit tests
+  - `security-check.js` - checks for security advisories in dependencies
+  - `update-json-schema.js` - helper for updating dependency version of vets-jason-schema
+  - `watch.js` - watch task for unit tests (proof of concept)
+- `/src` - source files for vets-website web applications
+  - `/applications` - application specific code that's into -entry.js files
+  - `/platform` - shared code that's bundled into vendor.js, injected directly into the site, or used for testing
+  - `/site` - build specific code including templates used for translating content from drupal CMS for the metalsmith build process
```
Comment (jbalboni) [inline packages/documentation/src/pages/getting-started/bundles-and-code-organization.mdx]: Maybe "Metalsmith pipeline code and templates used to generate static content pages from Drupal api or Markdown files"

File: packages/documentation/src/pages/getting-started/bundles-and-code-organization.mdx (hunk 2 above)
Comment (jbalboni) [reply packages/documentation/src/pages/getting-started/bundles-and-code-organization.mdx]: This does a static content build on Heroku after a vets-website branch is pushed to a Heroku instance when a PR is opened.

**Pull request merged**